### DISCUSSION TITLE
Dynamically added `Alpine.data()`

### DISCUSSION THREAD

**Discussion by notramo** [alpinejs/alpine#4232] [1. Help]
I found out that `Alpine.data()` can be called after `Alpine.start()`, the only requirement is that the markup that uses it has to be inserted after the data is defined:

1. Call `Alpine.start()`
2. Define data with `Alpine.data()`
3. Insert component into markup with [Turbo Streams](https://turbo.hotwired.dev/handbook/streams)

Since it is undocumented behavior, I'm curious, if it's intended or accidental.

Is it safe to use in large production apps? Is it planned to retain this behavior for future Alpine major versions (e.g. v4)?

**Comment by ekwoka** [comment]
There isn't any reason to lock down data, magic, directive, plugin, etc after start is called.

this can be useful for lazy loading components.
